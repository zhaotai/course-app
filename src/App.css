@import '~antd/dist/antd.css';

.app {
  text-align: center;
}

.cardContainer {
  padding: 16px;
}

.card {
  
}