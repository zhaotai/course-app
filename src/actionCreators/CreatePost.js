import dataLayer from "../dataLayer";

export function submitPost(values, cb) {
  dataLayer.createArticle(values).then(res => {
    cb();
  });
}