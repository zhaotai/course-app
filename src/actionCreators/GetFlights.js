import dataLayer from "../dataLayer";

export const Actions = {
  GetFlightsSuccess: "GetFlightsSuccess"
};

export function getFlights(dispatch) {
  setTimeout(() => {
    dataLayer.getFlights().then(res => {
      dispatch({
        type: Actions.GetFlightsSuccess,
        data: res
      });
      document.body.setAttribute("data-flights", res.flights.join(" "));
    });
  }, 1000);
}