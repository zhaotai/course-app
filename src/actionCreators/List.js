import dataLayer from "../dataLayer";

export const Actions = {
  GetListSuccess: "GetListSuccess"
};


export function loadList(dispatch) {
  dataLayer.getArticles().then(res => {
    dispatch({
      type: Actions.GetListSuccess,
      data: res.posts
    });
  });
}