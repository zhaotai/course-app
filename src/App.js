import React from "react";
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import List from "./pages/List";
import Details from "./pages/Details";
import CreatePost from "./pages/CreatePost/index";
import { connect } from "./stores";
import "./App.css";
import { getFlights } from "./actionCreators/GetFlights";

export const Pages = {
  List: "/",
  Details: "/details",
  Create: "/create"
};

class App extends React.Component {
  componentDidMount() {
    getFlights(this.props.store.dispatch);
  }

  render() {
    const flightsStore = this.props.store.getState().flightsStore;
    if (flightsStore.flights) {
      return (
        <Router>
          <div>
            <nav>
              <ul>
                <li>
                  <Link to={Pages.List}>Home</Link>
                </li>
              </ul>
            </nav>
            <Route path={Pages.List} exact render={routeProps => <List {...routeProps} {...this.props.store.getState()} dispatch={this.props.store.dispatch} />} />
            <Route path={`${Pages.Details}/:id`} exact render={routeProps => <Details {...routeProps} {...this.props.store.getState()} dispatch={this.props.store.dispatch} />} />
            <Route path={Pages.Create} exact component={CreatePost} />
          </div>
        </Router>
      );
    } else {
      return (
        <div>Loading</div>
      );
    }
  }
}

export default connect(App);
