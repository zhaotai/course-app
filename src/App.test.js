import React from 'react';
import ReactDOM from 'react-dom';
import Renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import * as GetFlights from './actionCreators/GetFlights';
import App from './App';

jest.mock('./stores');
GetFlights.getFlights = jest.fn();

describe("App test", () => {
  let store;
  beforeEach(() => {
    store = {
      getState: jest.fn(() => {
        return {
          flightsStore: {
            flights: null
          }
        }
      })
    };
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App store={store} />, div);
    expect(GetFlights.getFlights.mock.calls.length).toBe(1);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  it('render the whole tree', () => {
    const tree = Renderer
      .create(<App store={store} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  describe('shallow render', () => {
    let renderer;
    beforeEach(() => {
      renderer = new ShallowRenderer();
    });
  
    it('loading', () => {
      renderer.render(<App store={store} />);
      const result = renderer.getRenderOutput();
      expect(result).toMatchSnapshot();
    });
  
    it('list', () => {
      store.getState = jest.fn(() => {
        return {
          flightsStore: {
            flights: ["testFlight"]
          }
        }
      })
      renderer.render(<App store={store} />);
      const result = renderer.getRenderOutput();
      expect(result).toMatchSnapshot();
    });
  });
});
