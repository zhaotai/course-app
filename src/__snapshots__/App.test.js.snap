// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test render the whole tree 1`] = `
<div>
  Loading
</div>
`;

exports[`App test shallow render list 1`] = `
<BrowserRouter>
  <div>
    <nav>
      <ul>
        <li>
          <Link
            to="/"
          >
            Home
          </Link>
        </li>
      </ul>
    </nav>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/details/:id"
      render={[Function]}
    />
    <Route
      component={[Function]}
      exact={true}
      path="/create"
    />
  </div>
</BrowserRouter>
`;

exports[`App test shallow render loading 1`] = `
<div>
  Loading
</div>
`;
